src/common/get-file-info.js

```diff
@@ -67,27 +67,38 @@ function _getFileInfo({
 }) {
   const fileInfo = {
     ignored: ignorer.ignores(filePath),
-    inferredParser: options.inferParser(filePath, plugins) || null,
+    inferredParser: null,
   };
 
-  if (!fileInfo.inferredParser && resolveConfig) {
```

Comment:
Looks like a bug. `parser` can be overridden by the config.

Reply:
Yes, I noticed this.

Reply:
Added a test, I'll fix it in another PR.